packages/core-design-system/scripts/build.js

```diff
@@ -0,0 +1,287 @@
+import fs from 'node:fs/promises'
+
+import { permutateThemes, register } from '@tokens-studio/sd-transforms'
+import StyleDictionary from 'style-dictionary'
+import { getReferences, usesReferences } from 'style-dictionary/utils'
+
+import { harnessLog } from './complete-log.js'
+import {
+  COMMON_CONFIG,
+  DESIGN_SYSTEM_PREFIX,
+  DESIGN_SYSTEM_ROOT,
+  DESIGN_SYSTEM_ROOT_ESM,
+  getExportFileHeader,
+  STYLE_BUILD_FORMATS
+} from './constants.js'
+import { generateCoreFiles, generateThemeFiles } from './sd-file-generators.js'
+
+register(StyleDictionary)
+
+async function run() {
+  const $themes = JSON.parse(await fs.readFile('design-tokens/$themes.json'))
```

Comment:
Can we wrap this `JSON.parse` in try-catch?

Reply:
Updated